beets/dbcore/query.py

```diff
@@ -290,47 +298,57 @@ def string_match(cls, pattern: Pattern, value: str) -> bool:
         return pattern.search(cls._normalize(value)) is not None
 
 
-class BooleanQuery(MatchQuery):
+class BooleanQuery(MatchQuery[int]):
     """Matches a boolean field. Pattern should either be a boolean or a
     string reflecting a boolean.
     """
 
     def __init__(
             self,
             field: str,
-            pattern: Union[bool, str],
+            pattern: bool,
             fast: bool = True,
     ):
-        super().__init__(field, pattern, fast)
         if isinstance(pattern, str):
-            self.pattern = util.str2bool(pattern)
-        self.pattern = int(self.pattern)
+            pattern = util.str2bool(pattern)
 
+        pattern_int = int(pattern)
 
-class BytesQuery(MatchQuery):
+        super().__init__(field, pattern_int, fast)
+
+
+class BytesQuery(FieldQuery[bytes]):
     """Match a raw bytes field (i.e., a path). This is a necessary hack
     to work around the `sqlite3` module's desire to treat `bytes` and
     `unicode` equivalently in Python 2. Always use this query instead of
     `MatchQuery` when matching on BLOB values.
     """
 
     def __init__(self, field: str, pattern: Union[bytes, str, memoryview]):
-        super().__init__(field, pattern)
-
         # Use a buffer/memoryview representation of the pattern for SQLite
         # matching. This instructs SQLite to treat the blob as binary
         # rather than encoded Unicode.
-        if isinstance(self.pattern, (str, bytes)):
-            if isinstance(self.pattern, str):
-                self.pattern = self.pattern.encode('utf-8')
-            self.buf_pattern = memoryview(self.pattern)
+        if isinstance(pattern, (str, bytes)):
+            if isinstance(pattern, str):
+                bytes_pattern = pattern.encode('utf-8')
+            else:
+                bytes_pattern = pattern
+            self.buf_pattern = memoryview(bytes_pattern)
         elif isinstance(self.pattern, memoryview):
             self.buf_pattern = self.pattern
-            self.pattern = bytes(self.pattern)
+            bytes_pattern = bytes(self.pattern)
+        else:
+            raise ValueError("pattern must be bytes, str, or memoryview")
+
+        super().__init__(field, bytes_pattern)
```

Comment:
Another solid improvement to the old version, where the mutation was admittedly hard to follow.

Reply:
That's really just pleasing mypy, which does not like shadowing of variables at all.